sklearn/datasets/_base.py

```diff
@@ -264,7 +269,7 @@ def load_data(module_path, data_file_name):
         A 1D array containing the names of the classifications. For example
         target_names[0] is the name of the target[0] class.
     """
-    with open(join(module_path, "data", data_file_name)) as csv_file:
+    with importlib_resources.open_text(data_module, data_file_name) as csv_file:
```

Comment:
`.open_text` has `encoding="utf-8"` by default ([reference](https://github.com/python/cpython/blame/3.9/Lib/importlib/resources.py#L115-L118))

Happy to make it explicit though.

Reply:
I think this is fine to rely on the default utf-8 encoding assumption. I am pretty sure that all our CSV file only need the ascii encoding but this is a valid subset of utf-8 as far as I know.

We just needed to ensure that the loader would not implicitly try to use a platform dependent encoding.